contracts/beyondFramework/dca/libraries/InvestQueueLib.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+library InvestQueueLib {
+    uint256 public constant QUEUE_LEN = 244;
+
+    struct InvestQueue {
+        uint256[255] investAmounts;
+        uint8 current;
+    }
+
+    function getCurrentInvestmentAmountAndMoveNext(InvestQueue storage queue)
+        internal
+        returns (uint256)
+    {
+        uint256 amount = queue.investAmounts[queue.current];
+        queue.investAmounts[queue.current] = 0;
+        queue.current = _nextQueueIndex(queue.current);
+
+        return amount;
+    }
+
+    function splitUserInvestmentAmount(
+        InvestQueue storage queue,
+        uint256 amountToInvest,
+        uint8 amountSplit
+    ) internal {
+        require(
```

Comment:
Linter keep complaining about some of requires have too long error message. We can shorten their error message or add the following line at the above them. `// solhint-disable-next-line reason-string`